### PR TITLE
use "unique" check for certain network facts

### PR DESCRIPTION
**Pull request by beav**
Previously, we were raising an alert if two systems had different IP
addresses or HW addresses.

Instead, only raise an alert if we don't have unique addresses across
all systems compared.

### CODE REVIEW

File: tests/test_v1_api.py [hunk 1]
```diff
@@ -95,6 +95,15 @@ def test_comparison_report_api(self, mock_fetch_systems):
...
         self.assertEqual(response.status_code, 200)
+        comparisons = json.loads(response.data)
+        # we hard-code the index lookup since we know the fixture layout
+        network_comparisons = comparisons["facts"][12]["comparisons"]
+        ipv4_comparison = [
+            comparison
+            for comparison in network_comparisons
+            if comparison["name"] == "eth99.ipv4_addresses"
+        ][0]
+        self.assertEqual(ipv4_comparison["state"], "DIFFERENT")
```
Comment (ehelms) [inline tests/test_v1_api.py]: Could we add individual tests for each unique fact to ensure none of them regress?

File: drift/info_parser.py [hunk 2]
```diff
@@ -189,6 +203,23 @@ def _create_comparison(systems, info_name):
...
+    if not info_name.startswith("network_interfaces"):
+        return
+    elif info_name.startswith("network_interfaces.lo."):
+        return
+    elif info_name.endswith(".ipv4_addresses"):
+        return True
+    elif info_name.endswith(".ipv6_addresses"):
+        return True
+    elif info_name.endswith(".mac_address"):
+        return True
```
Comment (ehelms) [inline drift/info_parser.py]: We need the fqdn included as well as a unique recommendation

Comment (ehelms) [reply drift/info_parser.py]: Thoughts on a dictionary stored either in a separate python file or json/yaml object that maps property to its operator? And default to 'same' if no entry.

Comment (beav) [reply drift/info_parser.py]: I can split it out into an array which may help with readability

**Comment by beav** [comment]
@ehelms updated
